packages/trpc/server/routers/viewer/bookings/deletePastBookings.handler.ts

```diff
@@ -0,0 +1,48 @@
+import { prisma } from "@calcom/prisma";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+import type { TDeletePastBookingsSchema } from "./deletePastBookings.schema";
+
+type DeletePastBookingsOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TDeletePastBookingsSchema;
+};
+
+export const deletePastBookingsHandler = async ({ ctx, input }: DeletePastBookingsOptions) => {
+  const { user } = ctx;
+  const { bookingIds } = input;
+
+  const bookings = await prisma.booking.findMany({
+    where: {
+      id: { in: bookingIds },
+    },
+  });
+
+  const unauthorized = bookings.some((booking) => booking.userId !== user.id);
+  if (unauthorized) {
+    throw new Error("Unauthorized: Cannot delete bookings that don't belong to you");
```

Comment:
Which one? Queue bunch of debugging :)

Reply:
Also this does not support team bookings, should it?

Reply:
> Which one? Queue bunch of debugging :)

Right, I should mention bookingId as well, noted. 



> Also this does not support team bookings, should it?

This should also only allow admins delete bookings which are team bookings, I have missed that, will add that condition.